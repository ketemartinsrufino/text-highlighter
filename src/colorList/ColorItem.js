import { useState } from 'react';
import styled from 'styled-components';

const ColorItemStyled = styled.div`
    border: ${({isSelected}) => isSelected ? 2: 1}px black solid;
    height:30px;
    width: 50px;
    background-color: ${({color}) => color};

    :hover {
        cursor: pointer:
    }
`;

const ColorItem = ({color}) => {

    return <ColorItemStyled 
            color="red" 
            isSelected={this.isSelected("red")}
            onClick={this.getOnClick("red")}
        />
}