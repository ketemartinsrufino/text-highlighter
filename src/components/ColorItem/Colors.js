export const COLORS = {
  red: '#ef5350', 
  yellow: '#ffeb3b', 
  green: '#8eda8e'
}