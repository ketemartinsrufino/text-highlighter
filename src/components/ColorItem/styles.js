import styled from 'styled-components';

export const ColorItemStyled = styled.div`
  background-color: ${({color}) => color};
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ${({ isSelected }) => isSelected && `
    border: 2px black solid
  `}

  :hover {
    cursor: pointer:
  }
`;