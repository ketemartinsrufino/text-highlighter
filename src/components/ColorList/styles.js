import styled from 'styled-components';

export const ColorContainerStyled = styled.div`
  display: flex;
  margin: 10px 0;
`;
