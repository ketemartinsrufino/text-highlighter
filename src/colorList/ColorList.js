import React from 'react';
import styled from 'styled-components';
import ColorItem from './ColorItem';
import PropTypes from 'prop-types';

const ColorContainerStyled = styled.div`
    display: flex;
`;

const ColorList = ({
  colors,
  selectedColors, 
  onClick 
}) => {
  // console.log('ColorList', {
  //   colors,
  //   selectedColors, 
  //   onClick 
  // });

  return (
    <ColorContainerStyled>
      {colors.map( color => (
        <ColorItem
          key={`color-${color}`}
          color={color}
          isSelected={selectedColors.includes(color)}
          onClick={onClick}
        />
      ))}
    </ColorContainerStyled>
  )
}
ColorList.defaultProps = {
  colors: ['#ef5350', '#ffeb3b', '#8eda8e'], 
  selectedColors: []
}

ColorList.propTypes = {
  colors: PropTypes.array,
  selectedColors: PropTypes.array,
  onClick: PropTypes.func.isRequired
};

export default ColorList;