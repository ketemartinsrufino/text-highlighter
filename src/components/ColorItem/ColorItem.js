import React from 'react';
import PropTypes from 'prop-types';
import { ColorItemStyled } from './styles';

const ColorItem = ({ color, isSelected, onClick }) => (
  <ColorItemStyled
    color={color}
    isSelected={isSelected}
    onClick={() => onClick(color)}
    className={`ColorItem${isSelected ? ' selected' : ''}`}
  />
)

ColorItem.defaultProps = {
  isSelected: false
}

ColorItem.propTypes = {
  isSelected: PropTypes.bool,
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ColorItem;