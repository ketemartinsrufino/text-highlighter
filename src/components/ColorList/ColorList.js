import React from 'react';
import PropTypes from 'prop-types';
import { ColorContainerStyled } from './styles';
import ColorItem from '../ColorItem/ColorItem';

const ColorList = ({ colors, selectedColors, onClick }) => (
  <ColorContainerStyled>
    {colors.map( color => (
      <ColorItem
        key={`color-${color}`}
        color={color}
        isSelected={selectedColors.includes(color)}
        onClick={onClick}
      />
    ))}
  </ColorContainerStyled>
)
ColorList.defaultProps = {
  colors: ['#ef5350', '#ffeb3b', '#8eda8e'], 
  selectedColors: []
}

ColorList.propTypes = {
  colors: PropTypes.array,
  selectedColors: PropTypes.array,
  onClick: PropTypes.func.isRequired
};

export default ColorList;