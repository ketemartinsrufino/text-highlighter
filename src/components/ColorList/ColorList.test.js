import React from 'react';
import { shallow } from 'enzyme';
import ColorList from './ColorList';
import toJson from 'enzyme-to-json';
import { COLORS } from '../ColorItem/Colors';

describe('ColorList', () => {
  it('is defined with a default value', () => {
    const component = shallow(<ColorList onClick={jest.fn()}/>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('receive a list of "selectedColors"', () => {
    const component = shallow(
      <ColorList 
        selectedColors={[COLORS.green, COLORS.yellow]} 
        onClick={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('custom colors', () => {
    const component = shallow(
      <ColorList 
        colors={["purple", "blue"]} 
        onClick={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});