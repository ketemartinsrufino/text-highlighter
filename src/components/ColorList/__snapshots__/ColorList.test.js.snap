// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList custom colors 1`] = `
<styled.div>
  <ColorItem
    color="purple"
    isSelected={false}
    key="color-purple"
    onClick={[MockFunction]}
  />
  <ColorItem
    color="blue"
    isSelected={false}
    key="color-blue"
    onClick={[MockFunction]}
  />
</styled.div>
`;

exports[`ColorList is defined with a default value 1`] = `
<styled.div>
  <ColorItem
    color="#ef5350"
    isSelected={false}
    key="color-#ef5350"
    onClick={[MockFunction]}
  />
  <ColorItem
    color="#ffeb3b"
    isSelected={false}
    key="color-#ffeb3b"
    onClick={[MockFunction]}
  />
  <ColorItem
    color="#8eda8e"
    isSelected={false}
    key="color-#8eda8e"
    onClick={[MockFunction]}
  />
</styled.div>
`;

exports[`ColorList receive a list of "selectedColors" 1`] = `
<styled.div>
  <ColorItem
    color="#ef5350"
    isSelected={false}
    key="color-#ef5350"
    onClick={[MockFunction]}
  />
  <ColorItem
    color="#ffeb3b"
    isSelected={true}
    key="color-#ffeb3b"
    onClick={[MockFunction]}
  />
  <ColorItem
    color="#8eda8e"
    isSelected={true}
    key="color-#8eda8e"
    onClick={[MockFunction]}
  />
</styled.div>
`;
